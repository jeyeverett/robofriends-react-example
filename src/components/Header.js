import React, { Component } from 'react';

class Header extends Component {
    shouldComponentUpdate(nextProps, nextState) { //This prevents the header from rerendering
        return false;
    }
    render() {
        console.log('Header');
        return (
            <div>
                <h1 className='f1'>RoBros</h1>
            </div>
        );
    }
}

export default Header;