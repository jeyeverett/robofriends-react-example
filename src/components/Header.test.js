import { create } from "react-test-renderer";
import React from 'react';
import Header from './Header';

const header = create(<Header />);

it('expect to render Header component', () => {
    expect(header.toJSON()).toMatchSnapshot()
});

