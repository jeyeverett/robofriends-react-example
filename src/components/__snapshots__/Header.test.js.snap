// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Header component 1`] = `
<div>
  <h1
    className="f1"
  >
    RoBros
  </h1>
  <button
    id="counter"
    onClick={[Function]}
  >
    Count: 
    0
  </button>
</div>
`;
